test/state_test.js

```diff
@@ -217,24 +234,36 @@ describe('State', () => {
     assume(spotRequests).has.length(1);
     assume(spotRequests[0]).has.property('state', secondState);
   });
+   
+  it('should be able to update an AMI\'s usage', async () => {
+    await db.insertAmiUsage({region: defaultSR.region, id: defaultSR.id});
+    let amiUsage = await db.listAmiUsage(); 
+    assume(amiUsage).has.length(1);
+    
+    await db.updateAmiUsage({
+      region: 'us-west-1',
+      id: defaultSR.id,
+    });
+     
+    amiUsage = await db.listAmiUsage();
+    assume(amiUsage).has.length(1);
+    assume(amiUsage[0]).has.property('region', 'us-west-1');
+  });
 
-  it('should be able to do a spot request upsert', async () => {
-    let firstState = 'open';
-    let secondState = 'closed';
-    defaultSR.state = firstState;
+  it('should be able to upsert an AMI\'s usage', async () => {
 
-    await db.upsertSpotRequest(defaultSR);
-    let spotRequests = await db.listSpotRequests(); 
-    assume(spotRequests).has.length(1);
-    assume(spotRequests[0]).has.property('state', firstState);
+    await db.upsertAmiUsage({region: defaultSR.region, id: defaultSR.id});
+    let amiUsage = await db.listAmiUsage(); 
+    assume(amiUsage).has.length(1);
+    assume(amiUsage[0]).has.property('region', defaultSR.region);
 
-    defaultSR.state = secondState;
-    await db.upsertSpotRequest(defaultSR);
-    spotRequests = await db.listSpotRequests(); 
-    assume(spotRequests).has.length(1);
-    assume(spotRequests[0]).has.property('state', secondState);
+    let newRegion = 'us-west-1';
+    await db.upsertAmiUsage({region: newRegion, id: defaultSR.id});
+    amiUsage = await db.listAmiUsage(); 
+    assume(amiUsage).has.length(1);
+    assume(amiUsage[0]).has.property('region', 'us-west-1');
```

Comment:
What we really need to be checking for here is that when we use the same `region` and `id` that the timestamp stored in the database actually changes.  Both calls of `upsertAmiUsage` (which should be renamed `reportAmiUsage`) should be passed identical values.  I can't remember the resolution of the timestamps stored here, but we might need to insert a little delay to ensure that the database ticks at least one value over between the upsert calls.